nerfstudio/engine/trainer.py

```diff
@@ -137,6 +137,12 @@ def __init__(self, config: TrainerConfig, local_rank: int = 0, world_size: int =
 
         self.viewer_state = None
 
+        # used to keep track of the current step
+        self.step = 0
+        # A flag to stop training early for gradio webui, added seperately to avoid
+        # interference with viewer
+        self.early_stop = False
```

Comment:
Can we add a method like:
```
def shutdown(self) -> None:
    """Stop down the trainer and stop all associated threads/processes (such as the viewer)."""
```
that sets this to true? Usage-wise a method seems more self-documenting than an attribute that just needs to be overwritten